### PR TITLE
Adding in the ability to use ChatBot configurations

### PR DESCRIPTION
**Pull request by StewartW**
*Issue #, if available:*
#257 

*Description of changes:*
Enabled AWS ChatBot for pipeline notifications. 
Fixed a bug where multiple pipelines would result in a policy size exceeded error 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/schema_validation.py [hunk 1]
```diff
@@ -10,9 +10,14 @@
 
 LOGGER = configure_logger(__name__)
 
+NOTIFICATION_PROPS = {
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/schema_validation.py:13]: Could you update the documentation to explain how this should be setup too?

See: https://github.com/awslabs/aws-deployment-framework/blob/master/docs/user-guide.md#params

**Review by sbkok — CHANGES_REQUESTED** [review]
Thanks for providing this feature. A few small changes requested 👍

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/schema_validation.py (hunk 1 above)
Comment (StewartW) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/schema_validation.py:13]: Added in guidance as part of the user guide and admin guide

File: docs/admin-guide.md [hunk 2]
```diff
@@ -362,6 +362,19 @@ pipelines:
         name: omg_production
 ```
 
+### Integrating with Slack with AWS ChatBot
+The ADF also now supports integrating pipeline notifications with slack via the AWS ChatBot. This allows pipeline notifications to scale and provides a consistent slack notification across different AWS services. 
+
+In order to use AWS ChatBot, you must have already configured an (AWS ChatBot Client)[https://us-east-2.console.aws.amazon.com/chatbot/home?region=eu-west-1#/chat-clients] for your desired slack workspace. Once the client has been created. You will need to manually create a channel configuration that will be used by the ADF. 
+
+Currently, there isn't the ability to dynamically create channel configurations, but each deployment map can have a unique channel configured via the notification endpoint param.
```
Comment (sbkok) [inline docs/admin-guide.md:370]:
```suggestion
Currently, dynamically creating channel configurations is not supported. In the deployment map, you can configure a unique channel via the notification endpoint parameter for each pipeline separately. Add the `params` section if that is missing and add the following configuration to the pipeline:
```

File: docs/admin-guide.md [hunk 3]
```diff
@@ -362,6 +362,19 @@ pipelines:
...
 ```
 
+### Integrating with Slack with AWS ChatBot
+The ADF also now supports integrating pipeline notifications with slack via the AWS ChatBot. This allows pipeline notifications to scale and provides a consistent slack notification across different AWS services. 
+
+In order to use AWS ChatBot, you must have already configured an (AWS ChatBot Client)[https://us-east-2.console.aws.amazon.com/chatbot/home?region=eu-west-1#/chat-clients] for your desired slack workspace. Once the client has been created. You will need to manually create a channel configuration that will be used by the ADF. 
+
+Currently, there isn't the ability to dynamically create channel configurations, but each deployment map can have a unique channel configured via the notification endpoint param. 
+```
+    params:
```
Comment (sbkok) [inline docs/admin-guide.md:372]:
```suggestion
pipelines:
  - name: some-pipeline
    # ...
    params:
```

File: docs/user-guide.md [hunk 4]
```diff
@@ -195,8 +195,14 @@ Pipelines also have parameters that don't relate to a specific stage but rather
...
 
-- *notification_endpoint* *(String)* defaults to none.
-  > Can either be a valid email address or a string that represents the name of a Slack Channel. In order to integrate ADF with Slack see [Integrating with Slack](./admin-guide.md) in the admin guide. By Default, Notifications will be sent when pipelines Start, Complete or Fail.
+- *notification_endpoint* *(String) | (Dict) * defaults to none.
+  > Can either be a valid email address or a string that represents the name of a Slack Channel. 
+  > A more complex configuration can be provided to integrate with Slack via AWS ChatBot. 
+  > ``` 
+  > notification_endpoint:
+  >   type: chat_bot
+  >   target: example_slack_channel (This is the name of an existing slack channel configuration within AWS Chat Bot)
```
Comment (sbkok) [inline docs/user-guide.md:204]:
```suggestion
  > ```yaml
  > notification_endpoint:
  >   type: chat_bot
  >   target: example_slack_channel  # This is the name of an slack channel configuration you created within the AWS Chat Bot service. This needs to be created before you apply the changes to the deployment map.
  > ```
  >
```

File: docs/user-guide.md [hunk 5]
```diff
@@ -195,8 +195,14 @@ Pipelines also have parameters that don't relate to a specific stage but rather
...
+  > In order to integrate ADF with Slack see [Integrating with Slack](./admin-guide.md) in the admin guide. By Default, Notifications will be sent when pipelines Start, Complete or Fail.
```
Comment (sbkok) [inline docs/user-guide.md:205]:
```suggestion
  > In order to integrate ADF with Slack see [Integrating with Slack](./admin-guide.md#integrating-with-slack-with-aws-chatbot) in the admin guide. By default, notifications will be sent when pipelines Start, Complete, or Fail.
```

**Review by sbkok — APPROVED** [review]
LGTM, thank you!

**Review by thomasmcgannon — APPROVED** [review]
lgtm, thanks!

**Comment by deltagarrett** [comment]
LGTM!

**Review by deltagarrett — COMMENTED** [review]
LGTM

**Pull request merged**